samples/sample-terraform/README.md

```diff
@@ -1,19 +1,93 @@
-## Sample Terraform 
+# Terraform template
 
-### Deployment Map example
+## Overview
+
+This repository contains a module that manage the deployment of terraform code to multiple accounts and regions.
+The module consists of four build stages defined in the following file:
+
+- `buildspec.yml`: install the version of terraform specified in the pipeline configuration
+- `tf_scan.yml`: (optional) scans for vulnerabilities in the terraform code using the terrascan application. If vulnerabilities are found, it will fail and block further execution in the pipeline. It is recommended to enable this step in all ADF terraform pipelines.
+- `tf_plan.yml`: get the list of accounts from the organization and run a terraform plan
+- `tf_apply.yml`: get the list of accounts from the organization and run a terraform plan and apply
+
+An optional approval step could be added between plan and apply as shown in the pipeline definition below.
+
+## Parameters
+
+- TERRAFORM_VERSION: the terraform version used to deploy the resource
+- TARGET_ACCOUNTS: comma separated list of target accounts
+- TARGET_OUS: comma separated list of target leaf OUs (parent OUs are supported)
+- REGIONS: comma separated list of target regions. If this parameter is empty, the main ADF region is used.
+
+### Deployment procedure
+
+1. Add a sample-terraform pipeline in ADF `deployment-map.yml` as in the example:
 
 ```yaml
-  - name: my-terraform-example
-    default_providers:
-      source:
-        provider: codecommit
-        properties:
-          account_id: 1111111111111
-      deploy:
-        provider: codebuild
-        properties:
-          image: "STANDARD_4_0"
-    targets:
-      - properties:
-          spec_filename: my_test_spec.yml
+- name: sample-terraform
+  default_providers:
+    source:
+      provider: codecommit
+      properties:
+        account_id: 111111111111 # source account id
+    build:
+      provider: codebuild
+      properties:
+        environment_variables:
+          TERRAFORM_VERSION: "0.14.10" # terraform version
```

Comment:
Could you update this example to use the latest version of Terraform instead?
Users might copy the samples and not change the version number.

And could you document in the user guide which versions of Terraform are supported by this implementation, for example, `Terraform >= 0.14.10`.

Reply:
Added the lastest version 1.0.10